assignments/00-hello-rust/00-hello-rust/src/calculator.rs

```diff
@@ -1 +1,155 @@
+        "|" | "or" | "OR" => Ok(Operators::Or),
+        "^" | "xor" | "XOR" => Ok(Operators::Xor),
+        _ => Err("Could not recognize the operator".to_owned()),
+    }
+}
+
+/// Performs an operation on two numbers and returns the result
+/// as a u32
+///
+/// # Arguments
+///
+/// * `first_number`: u32 - The first number
+/// * `second_number`: u32 - The second number
+/// * `operator`: Operators - The operator to be used
+///
+/// # Returns
+///
+/// u32 - The result of the operation
+fn calculator(first_number: u32, second_number: u32, operator: Operators) -> u32 {
+    match operator {
+        Operators::And => first_number & second_number,
+        Operators::Or => first_number | second_number,
+        Operators::Xor => first_number ^ second_number,
+    }
+}
+
+/// Prints the result of an operation
+///
+/// # Arguments
+///
+/// * `first_number`: u32 - The first number
+/// * `second_number`: u32 - The second number
+/// * `operator`: Operators - The operator to be used
+/// * `result`: u32 - The result of the operation
+///
+/// # Returns
+///
+/// None
+fn print_output(first_number: u32, second_number: u32, operator: Operators, result: u32) {
+    match operator {
+        Operators::And => println!(
+            "The result of {} & {} is {}",
+            first_number, second_number, result
+        ),
+        Operators::Or => println!(
+            "The result of {} | {} is {}",
+            first_number, second_number, result
+        ),
+        Operators::Xor => println!(
+            "The result of {} ^ {} is {}",
+            first_number, second_number, result
+        ),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use test_case::test_case;
+
+    use crate::calculator::{calculator, parse_number, parse_operation, Operators};
+
+    #[test_case("0b0101", 5; "binary")]
+    #[test_case("1010", 1010 ; "decimal")]
+    #[test_case("0x10", 16 ; "hexadecimal")]
+    fn test_parse_number(number: &str, expected: u32) {
+        assert_eq!(parse_number(number), expected);
+    }
+
+    #[test_case("AND", Operators::And ; "Uppercase AND")]
+    #[test_case("and", Operators::And ; "Lowercase and")]
+    #[test_case("&", Operators::And ; "Ampersand")]
+    #[test_case("OR", Operators::Or ; "Uppercase OR")]
+    #[test_case("or", Operators::Or ; "OR")]
+    #[test_case("|", Operators::Or ; "Pipe")]
+    #[test_case("XOR", Operators::Xor ; "Uppercase XOR")]
+    #[test_case("xor", Operators::Xor ; "XOR")]
+    #[test_case("^", Operators::Xor ; "Caret")]
+    fn test_parse_operation(operator: &str, expected: Operators) {
+        assert_eq!(parse_operation(operator).unwrap(), expected);
+    }
+
+    #[test_case(2, 27, Operators::And, 2; "AND")]
+    #[test_case(248, 58, Operators::Or, 250; "OR")]
+    #[test_case(12, 32, Operators::Xor, 44; "XOR")]
+    fn test_calculator(first_number: u32, second_number: u32, operator: Operators, result: u32) {
+        assert_eq!(calculator(first_number, second_number, operator), result);
+    }
```

Comment:
nice job on the tests; for this last chunk I'd also make sure to explicitly test binary and hexadecimal inputs as well